### ISSUE TITLE
Safari - Issue with colors and opacity

### ISSUE THREAD

**Issue by florianbouvot** [tailwindlabs/tailwindcss#7887]
**What version of Tailwind CSS are you using?**

v3.0.23

**What build tool (or framework if it abstracts the build tool) are you using?**

laravel-mix v6.0.43

**What version of Node.js are you using?**

v17.7.1

**What browser are you using?**

Safari 15.4

**What operating system are you using?**

macOS 12.3

**Reproduction URL**

https://play.tailwindcss.com/UpPKozphrC

**Describe your issue**

**It's may be more a Safari bug than a Tailwind CSS bug.**

I create a button with an opacity to 0.9 on hover and button are not visible: every colors (text, bg, ...) seems to switch to black with alpha to 0 on Safari 15.4 (macOS).

**Comment by adamwathan** [comment]
Do you get the same problem when you just write regular CSS to do the same thing with no Tailwind at all? I tested on Safari Technology Preview and can't reproduce, but I haven't updated macOS to get 15.4 proper yet.

**Comment by florianbouvot** [comment]
In the following video I make some test and I identify that the bug is linked to the use of several property together:
- `border-radius` with a value greater than zero
- `border-color` transprent
- opacity use on hover

https://user-images.githubusercontent.com/2061540/159026713-2ae125cc-5ac2-4490-928e-932ed5807df6.mov

Code use in this video : https://play.tailwindcss.com/6xKNJkzw5Y

**Comment by Jxyme** [comment]
Hi,

I am able to reproduce the issues mentioned by @florianbouvot. 

I'm also witnessing some odd issues with `border-radius`, `border-color` and `opacity`, displayed here: [uploaded to Imgur](https://i.imgur.com/J9iTReD.mp4).

When updating macOS to 12.3 I immediately noticed issues with buttons that were using several properties together.

I did a few tests, gradually adding more properties and when `inline-flex` was added to the mix the issues occurred again.

I have included a playground link here which includes some examples for you: https://play.tailwindcss.com/3tMDZaFJg9.

The issues are also present on iOS 15.4 (I imagine iOS 15.4 = Safari 15.4 so it makes sense).

Hopefully the information provided helps - thank you in advance.

**Comment by thecrypticace** [comment]
This looks like a Safari 15.4 regression. I just tested in 15.3 and it's working fine but in 15.4 it fails. Another thing to note is if you remove the opacity the button background shows up again. I don't see any open webkit bugs for this yet though.

**Comment by thecrypticace** [comment]
Reduced test case:
```html
<button
  style="border-width: 2px;
    border-style: solid;
    background-color: rgb(255 0 0);
    color: black;
    border-radius: 0.375rem;
    opacity: 0.6;
    display: inline-flex;
    border-color: transparent;
  "
>
  Not working
</button>
```

Remove any property below color and the background appears.

**Comment by thecrypticace** [comment]
Alright I opened a bug with webkit for this: https://bugs.webkit.org/show_bug.cgi?id=238088

**Comment by adamwathan** [comment]
Going to close this as unfortunately not anything we can do about it — just a browser bug that affects everyone, regardless of if they are using Tailwind or writing CSS by hand. Hopefully something they will fix soon 🤞

**Comment by matheins** [comment]
We had the same issue with our buttons.
We solved that one by adding `-webkit-appearance: none;` to the buttons css. Hope this helps some of you!

**Comment by Jxyme** [comment]
> We had the same issue with our buttons.
> We solved that one by adding `-webkit-appearance: none;` to the buttons css. Hope this helps some of you!

This issue was resolved in MacOS 12.5 and iOS 15.6 released a few days ago to the public.

**Comment by yoonsikp** [comment]
Had a similar bug but it was fixed by adding position: relative;

https://stackoverflow.com/questions/73047620/safari-box-shadow-bug
